File: backend/__SLUG__/src/domains/health/service.go
package health

import "time"

// Service contains business logic for health domain
type Service struct {
	datasource *Datasource
}

// NewService creates a new service instance
func NewService(datasource *Datasource) *Service {
	return &Service{
		datasource: datasource,
	}
}

// GetHealth retrieves the current health status with business logic
func (s *Service) GetHealth() (*Health, error) {
	// Business logic: get system info from datasource
	service, version, err := s.datasource.GetSystemInfo()
	if err != nil {
		return nil, err
	}

	// Business logic: determine health status
	status := "ok"
	// Could add more sophisticated health checks here

	return &Health{
		Status:    status,
		Timestamp: time.Now().UTC(),
		Service:   service,
		Version:   version,
	}, nil
}

File: backend/__SLUG___node/src/domains/health/service.ts
import type { Datasource } from "./datasource";
import type { Health } from "./models";

export class Service {
  constructor(private readonly datasource: Datasource) {}

  getHealth(): Health {
    const { service, version } = this.datasource.getSystemInfo();
    return {
      status: "ok",
      timestamp: new Date().toISOString(),
      service,
      version,
    };
  }
}
